10-control-flow.Rmd

```diff
@@ -272,3 +272,67 @@ output_vector2
 > tests whether the country starts with a 'B', and graphs life expectancy
 > against time as a line graph if the mean life expectancy is under 50 years.
 >
+
+
+## Challenge solutions
+
+> ## Solution to Challenge 1 {.challenge}
+> We will first see a solution to Challenge 1 which does not use the `any` function.
+> We first obtain a logical vector describing which element of `gapminder$year` is equal to `2002`:
+> ```{r ch10pt1-sol, eval=FALSE}
+> gapminder[(gapminder$year == 2002),]
+> ```
+> Then, we count the number of rows of the data.frame `gapminder` that correspond to the 2002:
+> ```{r ch10pt2-sol, eval=FALSE}
+> rows2002_number <- nrow(gapminder[(gapminder$year == 2002),])
+> ```
+> The presence of any record for the year 2002 is equivalent to the request that `rows2002_number` is one or more:
+> ```{r ch10pt3-sol, eval=FALSE}
+> rows2002_number >= 1
+> ```
+> Putting all together, we obtain:
+> ```{r ch10pt4-sol, eval=FALSE}
+> if(nrow(gapminder[(gapminder$year == 2002),]) >= 1){
+>    print("Record found.")
+> }
+> ```
+
+> All this can be done more quickly with `any`. The logical condition can be expressed as:
+> ```{r ch10pt5-sol, eval=FALSE}
+> any(gapminder$year == 2002)
```

Comment:
I like the way you go into extended detail with this solution.
Can you expand this part of the solution to include an if statement and a 'pretty' message?
```
if(any(gapminder$year) == 2002){
    print("Record found.")
}
```